### ISSUE TITLE
Batchnorm Implemenatation?

### ISSUE THREAD

**Issue by shreyaskamathkm** [Orkis-Research/Pytorch-Quaternion-Neural-Networks#6]
Hi,

I was planning to test a few deep networks and was wondering if you have any implementation for the quaternion batch norm? Or any suggestion on how to implement it?
It would be of great help.

Thank you,
Best,
Shreyas Kamath

**Comment by Sahel13** [comment]
Hi,

I am also interested in this. Did you figure out a method?

**Comment by shreyaskamathkm** [comment]
@Sahel13 Please take a look at [this](https://github.com/Riccardo-Vecchi/Pytorch-Quaternion-Neural-Networks/blob/master/core/quaternion_layers.py).

**Comment by Sahel13** [comment]
@shreyaskamathkm That implementation of quaternion batchnorm looks quite different from the one introduced [here](https://arxiv.org/abs/1712.04604), which requires Cholesky decomposition. Do you know of a reference for that implementation?

**Comment by TParcollet** [comment]
Yes this one is different. The original one introduced in the papers simply is too computationally intensive. This implementation originate from a paper by D. Comminiello I believe.

**Comment by TParcollet** [comment]
My mistake, the reference is: Quaternion Convolutional Neural Network for
    Color Image Classification and Forensics

**Comment by Sahel13** [comment]
Thanks for the reference.
